source/llassetgen/include/llassetgen/Packing.h

```diff
@@ -0,0 +1,176 @@
+#pragma once
+
+#include <algorithm>
+#include <cstdint>
+#include <iterator>
+#include <type_traits>
+#include <vector>
+
+#include <llassetgen/Geometry.h>
+#include <llassetgen/llassetgen_api.h>
+
+namespace llassetgen {
+    using PackingSizeType = unsigned int;
+
+    /*
+     * Describes the packing of a texture atlas.
+     *
+     * The rectangles in `rects` correspond to the input rectangles at the same
+     * position in the packing algorithms input.
+     */
+    struct Packing {
+        Vec2<PackingSizeType> atlasSize{};
+        std::vector<Rect<PackingSizeType>> rects{};
+
+        Packing() = default;
+
+        LLASSETGEN_API explicit Packing(Vec2<PackingSizeType> _atlasSize);
+    };
+
+    namespace impl_packing {
```

Comment:
I suggest moving those internals away from the public header.

Reply:
Good idea, but there is one small problem: Several of the internal functions use `PackingSizeType` in their definitions, which is not an internal type alias and therefore should probably stay in `Packing.h`. Besides placing the alias before the include, my only idea would be to factor it out into a third header. Or just getting rid of the alias all together.